server/src/main/java/org/opensearch/extensions/action/ExtensionActions.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.action;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.extensions.DiscoveryExtension;
+import org.opensearch.extensions.ExtensionBooleanResponse;
+import org.opensearch.extensions.ExtensionsOrchestrator;
+import org.opensearch.extensions.RegisterTransportActionsRequest;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportException;
+import org.opensearch.transport.TransportResponse;
+import org.opensearch.transport.TransportResponseHandler;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * This class manages TransportActions for extensions
+ *
+ * @opensearch.internal
+ */
+public class ExtensionActions {
```

Comment:
We're really overusing the word "Action" throughout the codebase.  RestHandlers are "Actions" for example.  Per the class javadoc these are `TransportActions` so should the class name not align?

Reply:
Probably a two part answer. Half of it is here(above): https://github.com/opensearch-project/OpenSearch/pull/4598#discussion_r981759043
What would you recommend to name it ? 
Few options I can think of:
- ExtensionTransportActionsManager
- ExtensionActionsManager

Reply:
How about `ExtensionTransportActions`?

Reply:
I like `ExtensionTransportActions` but there's another class with that name below, so sure, toss on `Manager` or handler or whatever verb-based-noun applies!